grpcgcp/spanner_api_test.go

```diff
@@ -16,29 +16,196 @@
  *
  */
 
-package grpcgcp_test
+package grpcgcp
 
 import (
 	"context"
+	"crypto/rand"
 	"fmt"
+	"io"
+	"math/big"
+	"os"
+	"regexp"
+	"strconv"
 	"sync"
 	"testing"
+	"time"
 
 	"cloud.google.com/go/spanner"
-	"github.com/GoogleCloudPlatform/grpc-gcp-go/grpcgcp"
-	configpb "github.com/GoogleCloudPlatform/grpc-gcp-go/grpcgcp/grpc_gcp"
 	"google.golang.org/api/option"
 	"google.golang.org/grpc"
+
+	dbapi "cloud.google.com/go/spanner/admin/database/apiv1"
+	instapi "cloud.google.com/go/spanner/admin/instance/apiv1"
+	configpb "github.com/GoogleCloudPlatform/grpc-gcp-go/grpcgcp/grpc_gcp"
+	apb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+	ipb "google.golang.org/genproto/googleapis/spanner/admin/instance/v1"
+)
+
+var (
+	randId, _           = rand.Int(rand.Reader, big.NewInt(900000))
+	skipSpanner         = os.Getenv("SKIP_SPANNER")
+	gcpProjectId        = os.Getenv("GCP_PROJECT_ID")
+	spannerInstanceId   = "test-instance-" + strconv.FormatInt(100000+randId.Int64(), 10)
+	spannerInstancePath = fmt.Sprintf("projects/%s/instances/%s", gcpProjectId, spannerInstanceId)
+	spannerDbId         = "test-db"
+	database            = "projects/" + gcpProjectId + "/instances/" + spannerInstanceId + "/databases/" + spannerDbId
 )
 
 const (
-	database       = "projects/grpc-gcp/instances/sample/databases/benchmark"
-	tableName      = "storage"
-	column         = "id"
-	testSQL        = "select id from storage"
-	testColumnData = "payload"
+	regionName      = "regional-us-east1"
```

Comment:
Curious as to the reason to select this region. Maybe we can select green regions?

Reply:
Yeah, the region is not important. I will update it soon.